src/nvim/strings.c

```diff
@@ -561,3 +561,20 @@ char_u *concat_str(const char_u *restrict str1, const char_u *restrict str2)
   return dest;
 }
 
+// Reverse text into allocated memory.
+// Returns the allocated string.
+char_u *reverse_text(char_u *s) FUNC_ATTR_NONNULL_RET
```

Comment:
So `FUNC_ATTR_NONNULL_ALL` implies `FUNC_ATTR_NONNUL_RET` and `FUNC_ATTR_NONNUL_ARG` for all args?